src/drivers/imx/sai.c

```diff
@@ -355,6 +355,7 @@ static inline int sai_set_config(struct dai *dai, struct ipc_config_dai *common_
 		return -EINVAL;
 	}
 
+#ifndef CONFIG_IMX8ULP
 	switch (sai->params.tdm_slot_width) {
```

Comment:
If not supported by the current way the DMAs are configured the packing feature should just be removed altogether as it might break things if the slot width is not 32 (e.g: i.MX8ULP's case). IMO the right way would be to add it together with the support for compatible DMA configurations.

Reply:
As Daniel said, we would add a parameter for packing mode, then this can be controlled by the user.